optimum/habana/transformers/generation/utils.py

```diff
@@ -726,6 +726,9 @@ def generate(
         # determine whether flash attention needs to be used
         model_kwargs["use_flash_attention"] = generation_config.use_flash_attention
         model_kwargs["flash_attention_recompute"] = True if generation_config.flash_attention_recompute else False
+        model_kwargs["flash_attention_causal_mask"] = True if generation_config.flash_attention_causal_mask else False
+        model_kwargs["use_fused_rope"] = False if not generation_config.use_fused_rope else True
```

Comment:
Let's remove line 730 as `use_fused_rope` is mostly a workaround for the LoRA example at the moment

Reply:
Not needed, I removed it in second commit.